x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/RollupActionConfig.java

```diff
@@ -6,157 +6,205 @@
+            DateHistogramInterval fixedInterval = (DateHistogramInterval) a[0];
+            if (fixedInterval != null) {
+                return new RollupActionConfig(fixedInterval);
+            } else {
+                throw new IllegalArgumentException("Parameter [" + FIXED_INTERVAL + "] is required.");
+            }
         });
-        PARSER.declareObject(
-            optionalConstructorArg(),
-            (p, c) -> RollupActionGroupConfig.fromXContent(p),
-            new ParseField(RollupActionGroupConfig.NAME)
+
+        PARSER.declareField(
+            constructorArg(),
+            p -> new DateHistogramInterval(p.text()),
+            new ParseField(FIXED_INTERVAL),
+            ObjectParser.ValueType.STRING
         );
-        PARSER.declareObjectArray(optionalConstructorArg(), (p, c) -> MetricConfig.fromXContent(p), new ParseField(MetricConfig.NAME));
     }
 
-    public RollupActionConfig(final RollupActionGroupConfig groupConfig, final List<MetricConfig> metricsConfig) {
-        if (groupConfig == null && (metricsConfig == null || metricsConfig.isEmpty())) {
-            throw new IllegalArgumentException("At least one grouping or metric must be configured");
-        } else if (metricsConfig == null || metricsConfig.isEmpty()) {
-            throw new IllegalArgumentException("At least one metric must be configured");
+    /**
+     * Create a new {@link RollupActionConfig} using the given configuration parameters.
+     * @param fixedInterval the fixed interval to use for computing the date histogram for the rolled up documents (required).
+     */
+    public RollupActionConfig(final DateHistogramInterval fixedInterval) {
+        this.timeZone = DEFAULT_TIMEZONE;
+        if (fixedInterval == null) {
+            throw new IllegalArgumentException("Parameter [" + FIXED_INTERVAL + "] is required.");
         }
-        this.groupConfig = groupConfig;
-        this.metricsConfig = metricsConfig != null ? metricsConfig : Collections.emptyList();
+        this.fixedInterval = fixedInterval;
+
+        // validate interval
+        createRounding(this.fixedInterval.toString(), this.timeZone);
     }
 
     public RollupActionConfig(final StreamInput in) throws IOException {
-        groupConfig = in.readOptionalWriteable(RollupActionGroupConfig::new);
-        metricsConfig = in.readList(MetricConfig::new);
+        String intervalType = in.readString();
+        if (FIXED_INTERVAL.equals(intervalType) == false) {
+            throw new IllegalStateException("Invalid interval type [" + intervalType + "]");
+        }
+        fixedInterval = new DateHistogramInterval(in);
+        timeZone = in.readString();
     }
 
-    public RollupActionGroupConfig getGroupConfig() {
-        return groupConfig;
+    @Override
+    public void writeTo(final StreamOutput out) throws IOException {
+        out.writeString(FIXED_INTERVAL);
+        fixedInterval.writeTo(out);
+        out.writeString(timeZone);
```

Comment:
I chose to serialize timezone here because I thought we are going make it configurable in the future., so we better send a fixed string now. This will keep the protocol stable in the future. The same for interval type, I wanted to keep this information.

But you never know we may never find it useful to add support for multiple timezones and interval. We will definitely find something else to add there. So, for the sake of YAGNI principle, you think I should remove them?